Fixup notes (#41)
Sigmapitech/area-techno-tester: validation.md

Of all the frameworks we tried, React is the one where our group has the most experience. While Angular and
React are as complete as each other, our additional knowledge of the latter will make for a smoother development.
It is to be noted that while slightly harder than Vue, it retains its compatibility with capacitor and our
overall higher with it means that the difference is only relative. Thus, we decided to lean towards React as
the frontend for our web application.

## Backend technologies

For our backtend technologies, we will expand more on [fastAPI](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/back-fastapi), [Express](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/back-express) and [Flask](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/back-flask).

When it comes down to the backend, we let ourselves the choice between Express, Flask and FastAPI. Off the

> three we decided to go with FastAPI due to speed, the fact that it’s adapted to most API and our good experience with it.

Reviewer: ```suggestion
three we decided to go with FastAPI due to iteration speed, and its high versatility in constructing API. We also already build multiple projects with it, including the survivor jeb project.
```

It is also more structured than Express and less limited than Flask.

## Mobile technologies

For our backtend technologies, we will expand more on [React + capacitor](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/front-react), [Flutter](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/mobile-flutter) and [Kotlin](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/mobile-kotlin).

In the end our choice went to React combined with capacitor. While it doesn’t have the power of Flutter,

> nor the official label of Kotlin; our superior knowledge of it is a decent point in its favor.

Reviewer: "nor the official label of Kotlin" ?

Reviewer: "While it doesn’t have the first class support of Flutter,"

On top of that, the fact that it is React means that it will have a good synergy with our web frontend and
will potentially allow us to share some components. Additionally, the ease of installation is a very good point
for the different environment our team possesses.

## Databases

> For our backtend technologies, we will expand more on [MangoDB](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/db-mongo), [SQLAlchemy](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/db-sqlalchemy) and [SQLite](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/db-sqlite).

Reviewer: ```suggestion
For our backtend technologies, we have considered [MongoDB](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/db-mongo), [SQLAlchemy](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/db-sqlalchemy) and [SQLite](https://github.com/Sigmapitech/area-techno-tester/tree/main/considerations/db-sqlite).
```

For our database, we decided to lean towards SQLAlchemy. To begin with, it offers a better resilience

> than MangoDB, a very important factor for a database. We chose it over SQLite due to its scalability,

Reviewer: ```suggestion
than MongoDB, a very important factor for a database. We chose it over SQLite due to its scalability,
```
